honeybee_grasshopper_energy/src/HB Load Measure.py

```diff
@@ -0,0 +1,190 @@
+https://bcl.nrel.gov/nrel/types/measure
+-
+
+
+    Args:
+        _measure_path: Path to the folder in which the measure exists. This folder
+            must contain a measure.rb and a measure.xml file. Note that connecting
+            an input here will transform the component, essentially removing this
+            input and changing all of the other component inputs to be input
+            arguments for the measure.
+    
+    Returns:
+        measure: A measure measure object can be plugged into the "measures_" input
+            of the "HB Model To OSM" component in order to be included in the
+            export to OpenStudio.
+"""
+
+ghenv.Component.Name = 'HB Load Measure'
+ghenv.Component.NickName = 'LoadMeasure'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = 'HB-Energy'
+ghenv.Component.SubCategory = '5 :: Simulate'
+ghenv.Component.AdditionalHelpFromDocStrings = "3"
+
+try:
+    from honeybee_energy.measure import Measure
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee_energy:\n\t{}'.format(e))
+
+try:  # import the ladybug_rhino dependencies
+    from ladybug_rhino.grasshopper import all_required_inputs, give_warning
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug_rhino:\n\t{}'.format(e))
+
+import Grasshopper.Kernel as gh
+
+
+def add_component_input_from_arg(argument):
+    """Add an input parameter to this component using a MeasureArgument object.
+
+    Args:
+        argument: A honeybee-energy MeasureArgument object that will be used
+            to create a new input parameter for the component.
+    """
+    # create the input parameter object
+    param = gh.Parameters.Param_ScriptVariable()
+
+    # assign the required properties to the input
+    arg_id = '_{}'.format(argument.identifier) if argument.required else \
+        '{}_'.format(argument.identifier)
+    param.NickName = arg_id
+    param.Name = arg_id if argument.display_name is None else argument.display_name
+    param.AllowTreeAccess = False  # Measures don't support lists
+    param.Access = gh.GH_ParamAccess.item  # Measures don't support lists
+    param.Optional = True  # always needed so that default can come from measure file
+
+    # assign the optional properties to the input if they exist
+    if argument.type_text == 'Choice':
+        descr = [argument.description] if argument.description else []
+        descr.append('Choose from the following options:')
+        descr = descr + list(argument.valid_choices)
+        param.Description = '\n '.join(descr)
+    elif argument.description:
+        param.Description = argument.description
+    if argument.default_value is not None:
+        param.AddVolatileData(gh.Data.GH_Path(0), 0, argument.default_value)
+
+    # add the parameter to the compoent
+    index = ghenv.Component.Params.Input.Count
+    ghenv.Component.Params.RegisterInputParam(param, index)
+    ghenv.Component.Params.OnParametersChanged()
```

Comment:
Should we move this function out of the component and put it in the ladybug-rhino module?

Reply:
I think we can move all these related methods to a separate file as its own module.

Reply:
I understand that this is a long function but I am trying to be good at only putting stuff in the `ladybug_rhino` library that has the potential to be used in another component.  This code is so specific to how this component is set up and how it's supposed to transform that using this in another component will likely break that component (adding inputs that are not supposed to be there, overwriting the user's input to the component with other values) . So I would really like to keep these component-editing functions tied to this component if only to make it clear to people that these functions are not to be used outside this component. If there's something specific here that is potentially applicable to other components, let me know and I will move it.

Reply:
Reusability is important but more important than that is maintenance and interoperability between different tools. Since we use Grasshopper as the authoring tool for other possible virtual programming platforms we should keep all the plugin specific methods outside the components itself.

That way we can re-use the components between different platforms and only re-write the CAD dependent library. In this case ladybug-rhino.